tests/dependency_management/test_setupcfgt_writer.py

```diff
@@ -0,0 +1,302 @@
+    )
+
+    writer = SetupCfgWriter(store, tmpdir)
+    dependencies = [DefusedXML, Security]
+    changeset = writer.write(dependencies, dry_run=dry_run)
+
+    updated_setupcfg = """\
+        [metadata]
+        name = my_package
+        version = attr: my_package.VERSION
+
+        # some other stuff
+
+        [options]
+        include_package_data = True
+        python_requires = >=3.7
+        install_requires =
+            requests
+            importlib-metadata; python_version<"3.8"
+            defusedxml~=0.7.1
+            security~=1.2.0
+    """
+
+    assert setup_cfg.read() == (
+        dedent(orig_setupcfg) if dry_run else dedent(updated_setupcfg)
+    )
+
+    assert changeset is not None
+    assert changeset.path == setup_cfg.basename
+    res = (
+        "--- \n"
+        "+++ \n"
+        "@@ -10,3 +10,5 @@\n"
+        """ install_requires =\n"""
+        """     requests\n"""
+        """     importlib-metadata; python_version<"3.8"\n"""
+        """+    defusedxml~=0.7.1\n"""
+        """+    security~=1.2.0\n"""
+    )
+    assert changeset.diff == res
+    assert len(changeset.changes) == 2
+    change_one = changeset.changes[0]
+
+    assert change_one.lineNumber == 13
+    assert change_one.description == DefusedXML.build_description()
+    assert change_one.properties == {
+        "contextual_description": True,
+        "contextual_description_position": "right",
+    }
+    change_two = changeset.changes[1]
+    assert change_two.lineNumber == 14
+    assert change_two.description == Security.build_description()
+    assert change_two.properties == {
+        "contextual_description": True,
+        "contextual_description_position": "right",
+    }
+
+
+def test_add_same_dependency_only_once(tmpdir):
+    orig_setupcfg = """\
+        [metadata]
+        name = my_package
+        version = attr: my_package.VERSION
+
+        # some other stuff
+
+        [options]
+        include_package_data = True
+        python_requires = >=3.7
+        install_requires =
+            requests
+            importlib-metadata; python_version<"3.8"
+    """
+
+    setup_cfg = tmpdir.join("setup.cfg")
+    setup_cfg.write(dedent(orig_setupcfg))
+
+    store = PackageStore(
+        type=FileType.SETUP_CFG,
+        file=str(setup_cfg),
+        dependencies=set(),
+        py_versions=[">=3.7"],
+    )
+
+    writer = SetupCfgWriter(store, tmpdir)
+    dependencies = [Security, Security]
+    writer.write(dependencies)
+
+    updated_setupcfg = """\
+        [metadata]
+        name = my_package
+        version = attr: my_package.VERSION
+
+        # some other stuff
+
+        [options]
+        include_package_data = True
+        python_requires = >=3.7
+        install_requires =
+            requests
+            importlib-metadata; python_version<"3.8"
+            security~=1.2.0
+    """
+
+    assert setup_cfg.read() == dedent(updated_setupcfg)
+
+
+def test_dont_add_existing_dependency(tmpdir):
```

Comment:
We should have another test case where the version of `security` that already exists in the package is different from the one being added. There should be no new dependency added (or version changed) in this case either.

Reply:
This is a really good test case to think about. None of the other writers handle for this either so if it's ok with you it would be good to handle it in a separate PR for all. It would be worth thinking about it a bit more:  do we really not want to update the version? What if the project has `pkg==1.0` but our recommendation is `pkg==2.0`?

Worth thinking about it independently.